torch/_torch_docs.py

```diff
@@ -6446,3 +6446,126 @@ def merge_dicts(*dicts):
 `tensor([0, 0, ..., 1, 1, ..., 2, 2, ..., ...])` where `0` appears `n1` times,
 `1` appears `n2` times, `2` appears `n3` times, etc.
 """)
+
+
+add_docstr(torch._C.Generator,
+           r"""
+Generator(device='cpu', default=False) -> Generator
+
+Creates and returns a generator object which manages the state of the algorithm that
+produces pseudo random numbers. Used as a keyword argument in many random tensors in
+:ref:`inplace-random-sampling`. Currently only creation of CPU Generator is supported through
+this API.
+
+Arguments:
+    device (:class:`torch.device`, optional): the desired device for the generator.
+    default (bool, optional): set this to True to construct the default generator,
+    which will coincide with any other constructions of the default generator.
+
+Returns:
+    Generator: An ATen Generator object.
+
+Example::
+
+    >>> g_cpu = torch.Generator()
+    >>> g_cpu_default = torch.Generator(default=True);
+""")
+
+
+add_docstr(torch._C.Generator.set_state,
+           r"""
+Generator.set_state(new_state) -> void
+
+Sets the Generator state.
+
+Arguments:
+    new_state (torch.ByteTensor): The desired state.
+
+Example::
+
+    >>> g_cpu = torch.Generator()
+    >>> g_cpu_default = torch.Generator(default=True)
+    >>> g_cpu.set_state(g_cpu_default.get_state())
+""")
+
+
+add_docstr(torch._C.Generator.get_state,
+           r"""
+Generator.get_state() -> Tensor
+
+Returns the Generator state as a ``torch.ByteTensor``.
+
+Returns:
+    Tensor: A ``torch.ByteTensor`` which contains all the necessary bits
+    to restore a Generator to a specific point in time.
+
+Example::
+
+    >>> g_cpu = torch.Generator()
+    >>> g_cpu_default = torch.Generator(default=True)
+    >>> g_cpu.set_state(g_cpu_default.get_state())
+""")
+
+
+add_docstr(torch._C.Generator.manual_seed,
+           r"""
+Generator.manual_seed(seed) -> Generator
+
+Sets the seed for generating random numbers. Returns a `torch._C.Generator` object.
```

Comment:
You should refer to this as `torch.Generator` because that's the user visible name